File: threads/src/main/java/im/threads/internal/model/SearchingConsult.java
package im.threads.internal.model;

import androidx.core.util.ObjectsCompat;

import java.util.Calendar;
import java.util.Objects;

public final class SearchingConsult implements ChatItem {
    private long date;

    public SearchingConsult() {
        final Calendar c = Calendar.getInstance();
        c.set(Calendar.HOUR, 23);
        c.set(Calendar.MINUTE, 59);
        c.set(Calendar.SECOND, 59);
        this.date = c.getTimeInMillis();
    }

    public long getTimeStamp() {
        return date;
    }

    public void setDate(long date) {
        this.date = date;
    }

    @Override
    public boolean isTheSameItem(ChatItem otherItem) {
        return otherItem instanceof SearchingConsult;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        SearchingConsult that = (SearchingConsult) o;
        return date == that.date;
    }

    @Override
    public int hashCode() {
        return ObjectsCompat.hash(date);
    }

    @Override
    public String toString() {
        return "SearchingConsult{" +
                "date=" + date +
                '}';
    }
}

File: threads/src/main/java/im/threads/business/models/SearchingConsult.kt
package im.threads.business.models

import androidx.core.util.ObjectsCompat
import java.util.Calendar

class SearchingConsult : ChatItem {
    override var timeStamp: Long = 0

    init {
        val c = Calendar.getInstance()
        c[Calendar.HOUR_OF_DAY] = 23
        c[Calendar.MINUTE] = 59
        c[Calendar.SECOND] = 59
        timeStamp = c.timeInMillis
    }

    fun setDate(date: Long) {
        timeStamp = date
    }

    override fun isTheSameItem(otherItem: ChatItem?): Boolean {
        return otherItem is SearchingConsult
    }

    override val threadId: Long?
        get() = null

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as SearchingConsult
        return timeStamp == that.timeStamp
    }

    override fun hashCode(): Int {
        return ObjectsCompat.hash(timeStamp)
    }

    override fun toString(): String {
        return "SearchingConsult{" +
            "date=" + timeStamp +
            '}'
    }
}
